### PR TITLE
Circles 328 documentation

### PR DESCRIPTION
**Pull request by Peedee2002**
done a pass of the codebase to clean up some stuff

### CODE REVIEW

File: .github/workflows/ci_BE.yml [hunk 1]
```diff
@@ -34,7 +34,8 @@ jobs:
           echo "MONGO_INITDB_ROOT_USERNAME=github-ci" >> env/mongodb.env
           echo "MONGO_INITDB_ROOT_PASSWORD=github-ci" >> env/mongodb.env
           echo "REACT_APP_BACKEND_API_BASE_URL=http://localhost:8000/" >> env/frontend.env
-          docker compose run init-mongo
+          docker compose up -d backend
+          sleep 60
```
Comment (joooooooooooooooooooooooooooooooooooosh) [inline .github/workflows/ci_BE.yml]: bruh

Comment (Peedee2002) [reply .github/workflows/ci_BE.yml]: how else do i wait for the backend to start thooooo

**Review by joooooooooooooooooooooooooooooooooooosh — APPROVED** [review]

File: .github/workflows/ci_BE.yml (hunk 1 above)
Comment (Peedee2002) [reply .github/workflows/ci_BE.yml]: ok i came up with a less dumb solution

File: backend/server/routers/courses.py [hunk 2]
```diff
@@ -336,9 +336,30 @@ def unselectCourse(userData: UserData, unselectedCourse: str):
...
+        for course in courses_to_delete:
+            if user.has_taken_course(course):
+                user.pop_course(course)
+
+        courses_to_delete = [
+            c
+            for c in user.get_courses()
+            if CONDITIONS.get(c) is not None  # course is in conditions
+            and not (CONDITIONS[c].validate(user))[0]  # not unlocked anymore
+        ]
```
Comment (Joelrwest) [inline backend/server/routers/courses.py:359]: Why is courses_to_delete being constructed at the end and then nothing is done with it?

**Review by Joelrwest — APPROVED** [review]

File: backend/server/routers/courses.py (hunk 2 above)
Comment (Peedee2002) [reply backend/server/routers/courses.py:359]: its there to check if there are any new courses appended

Comment (Peedee2002) [reply backend/server/routers/courses.py:359]: while loops are crazy lol

Comment (Joelrwest) [reply backend/server/routers/courses.py:359]: Oh god I'm just dumb as and got the indenting wrong.... didn't realise it was in the while loop

**Pull request merged**